peepholelib/coreVectors/activations.py

```diff
@@ -88,33 +125,34 @@ def get_activations(self, **kwargs):
             continue
         
         # to check if pred and results data exist 
-        has_pred = 'pred' in cvs_td 
+        has_pred = 'pred' in act_td 
         
         # allocate memory for pred and result
         if not has_pred:
-            cvs_td['pred'] = MMT.empty(shape=torch.Size((n_samples,)))
-            cvs_td['result'] = MMT.empty(shape=torch.Size((n_samples,)))
+            act_td['output'] = MMT.empty(shape=torch.Size((n_samples,)+(num_classes,)))
+            act_td['pred'] = MMT.empty(shape=torch.Size((n_samples,)))
+            act_td['result'] = MMT.empty(shape=torch.Size((n_samples,)))
         
         # ---------------------------------------
         # compute predictions and get activations
         # ---------------------------------------
         
         # create a temp dataloader to iterate over images
-        cvs_dl = DataLoader(cvs_td, batch_size=bs, collate_fn = lambda x: x, shuffle=False) 
         act_dl = DataLoader(act_td, batch_size=bs, collate_fn = lambda x: x, shuffle=False) 
         
         if verbose: print('Computing activations')
         
-        for cvs_data, act_data in tqdm(zip(cvs_dl, act_dl), disable=not verbose, total=len(cvs_dl)):
+        for act_data in tqdm(act_dl, disable=not verbose, total=len(act_dl)):
             with torch.no_grad():
-                y_predicted = model(cvs_data['image'].to(device))
+                y_predicted = model(act_data['image'].to(device))
+                act_data['output'] = y_predicted
```

Comment:
this line ` act_data['output'] = y_predicted` should be inside the `if not has_pred:`